python_modules/dagster/dagster/_daemon/run_coordinator/queued_run_coordinator_daemon.py

```diff
@@ -279,6 +298,25 @@ def _get_runs_to_dequeue(
 
         return batch
 
+    def _notify_concurrency_blocked(
+        self, instance: DagsterInstance, run: DagsterRun, blocked_keys: Sequence[str]
+    ):
+        engine_events = instance.event_log_storage.get_logs_for_run(
+            run.run_id, of_type=DagsterEventType.ENGINE_EVENT, limit=1, ascending=False
+        )
+        last_event = engine_events[0] if engine_events else None
+        if (
+            not last_event
+            or "blocked waiting for op concurrency slots" not in last_event.message
+            or last_event.timestamp
+            < pendulum.now("UTC").timestamp() - CONCURRENCY_BLOCKED_MESSAGE_INTERVAL
+        ):
+            blocked_key_str = ", ".join(blocked_keys)
+            message = (
+                f"Run dequeue blocked waiting for op concurrency slots for keys: {blocked_key_str}"
+            )
+            instance.report_engine_event(message, run)
```

Comment:
I worry this could get really noisy without some debouncing, would maybe skip it for initial impl